src/Microsoft.Health.Fhir.Shared.Api/Features/Resources/Bundle/BundleHandler.cs

```diff
@@ -179,7 +184,16 @@ private async Task ExecuteRequests(Hl7.Fhir.Model.Bundle responseBundle, Hl7.Fhi
                     entryComponent.Response = new Hl7.Fhir.Model.Bundle.ResponseComponent { Status = ((int)HttpStatusCode.NotFound).ToString() };
                 }
 
-                responseBundle.Entry.Add(entryComponent);
+                if (responseBundle.Entry?.Any() != true)
+                {
+                    List<EntryComponent> entryComponents = new EntryComponent[entrySize].ToList();
+                    entryComponents[entry.Value] = entryComponent;
+                    responseBundle.Entry = entryComponents;
+                }
+                else
+                {
+                    responseBundle.Entry[entry.Value] = entryComponent;
```

Comment:
We can remove the else and the `entryComponents[entry.Value] = entryComponent;` the and always do `responseBundle.Entry[entry.Value] = entryComponent;`.

Reply:
It will not work because  if Entry is null or count is zero, I am creating a list with the rows to the length of entrySize and setting it to responseBundle. Then for the subsequent iteration only else part will be executed

Reply:
What I mean is it should be able to be simplified to the following:
```csharp
if (responseBundle.Entry?.Any() != true)
{
    responseBundle.Entry = new EntryComponent[_requestCount].ToList();
}

responseBundle.Entry[tuple.Item2] = entryComponent;
```

Reply:
got it, thanks for the clarification

Reply:
Not sure why my comments are not showing up:

This does not need to happen in the foreach loop right? It could just happen once before the loop? I think it might be a little easier to read if:

1. Create an array of size _requestCount before foreach loop
2. Sets the entryComponent to the corresponding index in foreach loop
3. After the forech loop, either do responseBundle.Entry.AddRange(array) or do array.ToList()